### DISCUSSION TITLE
File upload & model path in one line

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#36825] [Ideas] [answered]
Here is a [trait](https://github.com/dillingham/interacts-with-uploads) Im using to create a file name and append it to the `validated()`.

Makes adding **public** file uploads that are only attached to a model very clean:
```php
public function store(CreateProfileRequest $request)
{
    $request->upload('banner');

    $profile = Profile::create($request->validated());

    return redirect()->route('profiles.show', $profile);
}
```
The image path is saved on the model:
```html
<img src="{{ $profile->banner }}" />
```

**Comment by jonatandorozco** [comment]
What do you think about using a trait for this?

**Reply by dillingham** [reply]
@jonorozcoc I think thats a great idea, I'll work on it tonight

**Reply by dillingham** [reply, marked as answer]
https://github.com/dillingham/interacts-with-uploads

Initial attempt works. Will add tests later

**Reply by dillingham** [reply]
Tests added
